spark3/src/test/java/org/apache/iceberg/actions/TestCreateActions.java

```diff
@@ -447,6 +447,30 @@ public void testProperties() throws Exception {
     }
   }
 
+  @Test
+  public void testSparkTableReservedProperties() throws Exception {
+    String destTableName = "iceberg_reserved_properties";
+    String source = sourceName("test_reserved_properties_table");
+    String dest = destName(destTableName);
+    createSourceTable(CREATE_PARQUET, source);
+    assertMigratedFileCount(Actions.snapshot(source, dest), source, dest);
+    SparkTable table = loadTable(dest);
+
+    String[] keys = {"provider", "format", "current-snapshot-id", "location"};
+
+    for (String entry : keys) {
+      Assert.assertTrue(
+              "Created table missing reserved property " + entry, table.properties().containsKey(entry));
+    }
+
+    Assert.assertEquals("Property value is not the expected value", "iceberg", table.properties().get("provider"));
+    Assert.assertEquals("Property value is not the expected value", "iceberg/parquet",
```

Comment:
I think if we make the error message shorter, this would fit on one line.

```
    Assert.assertEquals("Unexpected provider", "iceberg", table.properties().get("provider"));
    Assert.assertEquals("Unexpected format", "iceberg/parquet", table.properties().get("format"));
    ...
```